### ISSUE TITLE
TypeError: Cannot create property 'hooks' on string ???

### ISSUE THREAD

**Issue by rabin999** [sequelize/sequelize#10702]
During fetch from User model I got this

```
options.hooks = options.hooks === undefined ? false : Boolean(options.hooks);
                  ^

TypeError: Cannot create property 'hooks' on string 'food_id'
    at Function.<anonymous> (C:\Projects\learnSequelize\node_modules\sequelize\lib\associations\mixin.js:98:19)
    at Function.Order.associate (C:\Projects\learnSequelize\database\models\order.js:17:11)
    at Object.keys.forEach.modelName (C:\Projects\learnSequelize\database\models\index.js:30:19)
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (C:\Projects\learnSequelize\database\models\index.js:28:17)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (C:\Projects\learnSequelize\app.js:1:76)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
```

My User Models:

```
'use strict';
module.exports = (sequelize, DataTypes) => {
  const User = sequelize.define('User', {
    name: DataTypes.STRING,
    address: DataTypes.STRING,
    phone: DataTypes.STRING,
    email: DataTypes.STRING,
    password: DataTypes.STRING,
  }, {
    tableName: 'users'
  });
  User.associate = function(models) {
    User.belongsToMany(models.Role, { through: 'user_role', foreignKey: 'user_id', otherKey: 'role_id' });
    User.hasMany(models.Order, 'user_id');
    User.hasMany(models.Food, 'canteen_manager_id');
  };
  return User;
};

```

Food Model:
```
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Food = sequelize.define('Food', {
    title: DataTypes.STRING,
    price: DataTypes.FLOAT,
    quantity: DataTypes.SMALLINT,
    description: DataTypes.STRING,
    canteen_manager_id: DataTypes.INTEGER
  }, {
    tableName: 'foods'
  });
  Food.associate = function(models) {
    Food.belongsTo(models.User, 'canteen_manager_id');
  };
  return Food;
};
```

Constraint:
```
'use strict';

module.exports = {
  up: (queryInterface, Sequelize) => {
    return queryInterface.addConstraint('foods', ['canteen_manager_id'], {
      type: 'foreign key',
      name: 'fk_canteen_manager_id',
      references: {
        table: 'users',
        field: 'id'
      },
      onDelete: 'cascade',
      onUpdate: 'cascade'
    });
  },

  down: (queryInterface, Sequelize) => {
    return queryInterface.removeConstraint('foods', 'fk_canteen_manager_id');
  }
};

```

**Comment by SimonSchick** [comment]
The error occured in the `Order.js` file, which you have not included in this issue.

You have to give us a full reproducible code sample (ideally a single file).

**Comment by rabin999** [comment]
This is my Order Model:

```
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Order = sequelize.define('Order', {
    food_id: DataTypes.INTEGER,
    user_id: DataTypes.INTEGER,
    quantity: DataTypes.MEDIUMINT,
    unit_price: DataTypes.FLOAT,
    total_amount: DataTypes.FLOAT,
    status: {
      type:   DataTypes.ENUM,
      values: ['pending', 'completed', 'canceled']
    }
  }, {
    tableName: 'orders'
  });
  Order.associate = function(models) {
    Order.belongsTo(models.Food, 'food_id');
    Order.belongsTo(models.User, 'user_id');
  };
  return Order;
};
```

**Comment by SimonSchick** [comment]
I still don't know what version of sequelize you are using or anything else.
Please fill out the issue template.

**Comment by rabin999** [comment]
Sorry, this is my package json file

```
{
  "name": "learnSequelize",
  "version": "1.0.0",
  "description": "",
  "main": "app.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Rabin Bhandari",
  "license": "ISC",
  "dependencies": {
    "mysql2": "^1.6.5",
    "sequelize": "^5.2.12"
  }
}

```

**Comment by SimonSchick** [comment]
Oh, you can't pass a string as the second argument to `belongsTo`, you probably meant to use `{ as: <string> }`.

**Comment by rabin999** [comment]
Thank you so much for helping me out, it helped.
Documentation is hard to follow, I got stocked by documentation flow.

**Comment by SimonSchick** [comment]
@rabin999 you are free to create a PR to improve the documentation as you see fit, PRs are welcome.
I also suggest you may switch to TS as it will help you avoid these types of simple mistakes from the get go.
